src/domain/bitcoin/onchain/index.types.d.ts

```diff
@@ -111,3 +111,17 @@ interface IOnChainService {
     targetConfirmations,
   }: PayToAddressArgs): Promise<OnChainTxHash | OnChainServiceError>
 }
+
+interface OnChainEvent {
+  payload: string
+  sequence: string
+  recorded_at: number
+}
```

Comment:
should this be part of domain?

Reply:
We talked about this and the events themselves are part of the translation / adaptor layer. Its the handler that calls the use cases which is where we enter the app layer (and thus domain).

Reply:
Yup, lots of refactoring around this in subsequent branches